### PR TITLE
Add com.obsproject.Studio.V4L2Sink

### PR DESCRIPTION
**Pull request by sirn**
This pull request add `V4L2Sink` extension to `com.obsproject.Studio`.

Please see https://github.com/flathub/com.obsproject.Studio/pull/69 for more details.

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build com.obsproject.Studio.V4L2Sink

**Comment by LeandroStanger** [comment]
bot, build com.obsproject.Studio.V4L2Sink

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64", "i386"]
```
Comment (LeandroStanger) [inline flathub.json:2]:
```suggestion
  "only-arches": ["x86_64"]
```

**Comment by LeandroStanger** [comment]
bot, build com.obsproject.Studio.V4L2Sink

**Comment by LeandroStanger** [comment]
bot, build com.obsproject.Studio.V4L2Sink

File: com.obsproject.Studio.V4L2Sink.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (TingPing) [inline com.obsproject.Studio.V4L2Sink.metainfo.xml:1]: You have to actually install this file to `/app/share/metainfo`

File: com.obsproject.Studio.V4L2Sink.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://github.com/CatxFish/obs-v4l2sink/archive/1ec3c8ada0e1040d867ce567f177be55cd278378.tar.gz",
+          "sha256": "381f9a4346aa5d926a07116160957b38fb35c89173f441180b7c25b84c9463e5"
+        },
+        {
+          "type": "archive",
+          "//": "It needs a copy of the source for some cmake files, its easier to just download again",
```
Comment (TingPing) [inline com.obsproject.Studio.V4L2Sink.json:29]:
```suggestion
          "//": "It needs a copy of the source for some cmake files",
```

That specific comment was related to plugins in-tree

File: com.obsproject.Studio.V4L2Sink.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+<component type="addon">
+  <id>com.obsproject.Studio.V4L2Sink</id>
+  <extends>com.obsproject.Studio</extends>
+  <name>V4L2Sink Plugin</name>
+  <summary>Adds support for V4L2Sink</summary>
+  <url type="homepage">https://github.com/CatxFish/obs-v4l2sink</url>
+  <project_license>GPL-2.0</project_license>
+  <metadata_license>CC0-1.0</metadata_license>
+  <update_contact>sirn_AT_ogsite.net</update_contact>
```
Comment (hfiguiere) [inline com.obsproject.Studio.V4L2Sink.metainfo.xml:10]: I would recommend adding a release tag in there with a version.

**Comment by Bleuzen** [comment]
Hi @sirn, is there any progress on this? Would love to see it on flathub.

**Comment by sirn** [comment]
@Bleuzen not much right now, I might be able to find some time this month to work on it (fingers crossed)

**Comment by sirn** [comment]
Finally got some time to work on this. Not sure if this is the right way of doing things, but I've added `release` tags, setting version based on `git describe --tags`, and install metainfo file. Please advice :pray:

**Comment by LeandroStanger** [comment]
bot, build com.obsproject.Studio.V4L2Sink

**Comment by Bleuzen** [comment]
Is this still needed? I think OBS 26.1 has such a feature built in now, so this plugin seems obsolete.

**Comment by peteruithoven** [comment]
I've got 26.1.0 installed through Flatpak, I don't see a start virtual camera button. 
[The release notes](https://github.com/obsproject/obs-studio/releases/tag/26.1.0) still note "requires v4l2loopback-dkms"

Looks like it's missing a 128x128 icon? It looked in: 
`builddir/*/share/app-info/icons/flatpak/128x128/com.obsproject.Studio.V4L2Sink.png`

**Comment by sirn** [comment]
@Bleuzen is right; this is now included in 26.1. @peteruithoven V4L2Sink also requires v4l2loopback-dkms there's actually no actually advantage of using it over the built-in one. Though I also unable to get Virtual Camera button to show up even with v4l2loopback-dkms installed, but I'll try figure that out in the OBS Studio repo instead.

Sorry for the noise! Closing this PR...

**Pull request closed without merging**